src/crypto/SecretStorage.js

```diff
@@ -92,13 +94,13 @@ export class SecretStorage extends EventEmitter {
         switch (algorithm) {
         case SECRET_STORAGE_ALGORITHM_V1_AES:
         {
-            const decryption = new global.Olm.PkDecryption();
-            try {
-                if (opts.passphrase) {
-                    keyData.passphrase = opts.passphrase;
-                }
-            } finally {
-                decryption.free();
+            if (opts.passphrase) {
+                keyData.passphrase = opts.passphrase;
+            }
+            if (opts.key) {
+                const {iv, mac} = await encryptAES(ZERO_STR, opts.key, "");
```

Comment:
I guess this is the part that we won't have for existing keys, which leads to your potential migration question...?

Not quite sure what "style" of migration you had in mind for this one... Could we potentially do this one "silently": as in we would not toast or prompt about it, but just whenever we next happen to unlock 4S successfully and notice that this extra data is missing, we could add it in for the future...?

I am very curious what @dbkr thinks about this as well.

Reply:
Yeah, we could either do an upgrade the next time the user enters their passphrase, or we could do the whole toast-and-prompt dance.

Reply:
Nicest would probably be if we checked the key next time we did a decryption, adding this info if it succeeded. This would still presumably end up with secrets encrypted with 2 different keys if we added a secret before doing this and the user got their passphrase wrong? But also presumably this is what happens now?